test/upload-test-images.sh

```diff
@@ -16,19 +16,26 @@
 
 set -o errexit
 
+function upload_test_images() {
+  echo ">> Publishing test images"
+  local image_dirs="$(find $(dirname $0)/test_images -mindepth 1 -maxdepth 1 -type d)"
+  local docker_tag=$1
+
+  for image_dir in ${image_dirs}; do
+      local image_name="$(basename ${image_dir})"
+      local image="github.com/knative/serving/test/test_images/${image_name}"
+      ko publish -B ${image}
+      if [ -n "${docker_tag}" ]; then
+          image=${KO_DOCKER_REPO}/${image_name}
+          local digest=$(gcloud container images list-tags --filter="tags:latest" --format='get(digest)' ${image})
+          echo "Tagging ${image}@${digest} with ${docker_tag}"
+          gcloud -q container images add-tag ${image}@${digest} ${image}:${docker_tag}
```

Comment:
IIRC this tagging logic was supposed to work without any dependence on gcloud. We might as well wait for ko to support tags though and then strip this logic altogether.

Reply:
PRs replacing gcloud with docker are welcome. ;) And I'm looking forward to ko supporting tags so we can simplify many scripts we have. :D